R/calc_mutation_frequency_bin_region.R

```diff
@@ -0,0 +1,277 @@
+#' @param return_count Boolean statement to return mutation count per window (TRUE) or binary mutated/unmutated status (FALSE). Default is TRUE.
+#' @param drop_unmutated Boolean for whether to drop windows with 0 mutations. Only effective with "long" return format.
+#' @param from_indexed_flatfile Set to TRUE to avoid using the database and instead rely on flat-files (only works for streamlined data, not full MAF details). Default is TRUE.
+#' @param mode Only works with indexed flat-files. Accepts 2 options of "slms-3" and "strelka2" to indicate which variant caller to use. Default is "slms-3".
+#'
+#' @return Either a matrix or a long tidy table of counts per window.
+#'
+#' @rawNamespace import(data.table, except = c("last", "first", "between", "transpose"))
+#' @import dplyr tidyr GAMBLR.helpers
+#' @export
+#'
+#' @examples
+#' chr11_mut_freq = calc_mutation_frequency_bin_region(region = "chr11:69455000-69459900",
+#'                                                          slide_by = 10,
+#'                                                          window_size = 10000)
+#'
+calc_mutation_frequency_bin_region <- function(region,
+                                          chromosome,
+                                          start_pos,
+                                          end_pos,
+                                          these_samples_metadata = NULL,
+                                          these_sample_ids = NULL,
+                                          this_seq_type = c("genome", "capture"),
+                                          maf_data = NULL,
+                                          projection = "grch37",
+                                          slide_by = 100,
+                                          window_size = 1000,
+                                          return_format = "long",
+                                          min_count_per_bin = 0,
+                                          return_count = TRUE,
+                                          drop_unmutated = FALSE,
+                                          from_indexed_flatfile = TRUE,
+                                          mode = "slms-3") {
+  # Create objects to describe region both as string and individual objects
+  try(if (missing(region) & missing(chromosome)) {
+    stop("No region information provided. Please provide a region as a string in the chrom:start-end format, or as individual arguments. ")
+  })
+
+  if ((drop_unmutated | min_count_per_bin > 0) & return_format == "wide") {
+    message("To return a wide table, all samples and windows must be kept. Ignoring drop_unmutated and min_count_per_bin arguments. ")
+  }
+
+  if (missing(region)) {
+    region <- paste0(
+      chromosome, ":", start_pos, "-",
+      end_pos
+    )
+  } else {
+    chunks <- GAMBLR.helpers:::region_to_chunks(region)
```

Comment:
This is probably a typo. It should be `:::` instead of `::` because `region_to_chunks` is an exported object from GAMBLR.helpers, instead of internal.